packages/sdk/src/client.ts

```diff
@@ -664,6 +668,48 @@ export function SoundClient({
     return mintSchedules.flat().sort((a, b) => a.startTime - b.startTime)
   }
 
+  async function getContractError(txHash: string) {
+    if (
+      txHash === NULL_BYTES32 ||
+      txHash.slice(0, 2) !== '0x' ||
+      // Tx hash is 32 bytes, which is 64 hex characters + '0x'
+      txHash.length !== 66
+    ) {
+      throw new InvalidTxHashError({ txHash })
+    }
+
+    const provider = await _requireProvider()
+    const tx = await provider.getTransaction(txHash)
+
+    try {
+      // Simulate the original transaction
+      const response = await provider.call(
+        {
+          to: tx.to,
+          from: tx.from,
+          nonce: tx.nonce,
+          gasLimit: tx.gasLimit,
+          gasPrice: tx.gasPrice,
+          data: tx.data,
+          value: tx.value,
+          chainId: tx.chainId,
+          type: tx.type ?? undefined,
+          accessList: tx.accessList,
+        },
+        tx.blockNumber,
+      )
```

Comment:
The only way to know why a transaction failed after it was mined is to simulate the original call at the block it was mined on. The reason is because nothing is stored or logged about the error unless specifically implemented in the contract.